src/app/file-tree/file-tree.component.html

```diff
@@ -8,15 +8,21 @@
       <!-- The already selected file -->
       <button *ngIf="!node.child && data.selectedFile.absolutePath === node.absolutePath" mat-raised-button disabled>
         <mat-icon class="mr-2" [attr.aria-label]="'file icon'">description</mat-icon>{{ node.name }}
+        <span *ngIf="isPrimaryDescriptor(node.absolutePath)" class="bubble preview-bubble" data-cy="primary-descriptor-bubble"
+          >Primary</span
+        >
       </button>
       <!-- The not-selected files -->
       <button (click)="selectFile(node)" *ngIf="!(!node.child && data.selectedFile.absolutePath === node.absolutePath)" mat-stroked-button>
         <mat-icon color="primary" class="mr-2" [attr.aria-label]="'file icon'">description</mat-icon>{{ node.name }}
+        <span *ngIf="isPrimaryDescriptor(node.absolutePath)" class="bubble preview-bubble" data-cy="primary-descriptor-bubble"
```

Comment:
Note: To ensure that this ticket gets through the review process, I do not recommend that the implementer use the following technique.  I mention it every so often because I think we could harness it to simplify our UI code...

A while back, I did some research on some generic pipes that would simultaneously allow functions to be invoked in templates and minimize the number of invocations by harnessing the Angular change detection logic: https://github.com/dockstore/dockstore-ui2/pull/1510

In this case, the code above would become something like:
`<span *ngIf="node.absolutePath | function: this:isPrimaryDescriptor"`

I like it because what's happening isn't buried in the typical pipe infrastructure, it's more flexible, and we don't end up with a million overly-specialized pipes.

Reply:
[This particular comment is a dummy because github won't let me submit a "Comment" review until I include it]